### ISSUE TITLE
Fix `use-set-literal` lint errors

### ISSUE THREAD

**Issue by harupy** [mlflow/mlflow#6535]
Fix `use-set-literal` lint errors by following the instructions below (example PR: #6534):

https://github.com/mlflow/mlflow/blob/1b89add428da0a4453e7523a20d13f04f6291f37/pylint_plugins/set_checker.py#L12-L41

## What needs to be fixed?

`set` applied on `list` or `tuple` literal need to be converted to `set` literal:

```python
# Bad
set([1, 2, 3])
set((1, 2, 3))

# Good
{1, 2, 3}
```

## Files

| File                                                               | Assignee     | PR    |
| :----------------------------------------------------------------- | :----------- | :---- |
| mlflow/azure/client.py                                             | @Monkero     | #6544 |
| mlflow/entities/lifecycle_stage.py                                 | @Monkero     | #6544 |
| mlflow/entities/run_status.py                                      | @Monkero     | #6544 |
| mlflow/server/handlers.py                                          | @dsgibbons   | #6542 |
| mlflow/tracking/fluent.py                                          | @dsgibbons   | #6542 |
| mlflow/utils/autologging_utils/safety.py                           | @dsgibbons   | #6542 |
| tests/autologging/test_autologging_behaviors_unit.py               | @dsgibbons   | #6543 |
| tests/entities/test_run_info.py                                    | @dsgibbons   | #6543 |
| tests/entities/test_run_status.py                                  | @dsgibbons   | #6543 |
| tests/lightgbm/test_lightgbm_autolog.py                            | @iamthen0ise | #6560 |
| tests/models/test_model_input_examples.py                          | @iamthen0ise | #6560 |
| tests/projects/test_project_spec.py                                | @iamthen0ise | #6560 |
| tests/pyfunc/test_model_export_with_loader_module_and_data_path.py |              |       |
| tests/store/artifact/test_databricks_artifact_repo.py              |              |       |
| tests/store/artifact/test_ftp_artifact_repo.py                     |              |       |
| ~tests/tensorflow/test_tensorflow2_autolog.py~                     |              |       |
| tests/utils/test_requirements_utils.py                             |              |       |
| tests/xgboost/test_xgboost_autolog.py                              |              |       |
| mlflow/utils/search_utils.py                                       | @harupy      | #6534 |
| tests/entities/model_registry/test_model_version.py                | @harupy      | #6534 |
| tests/entities/model_registry/test_registered_model.py             | @harupy      | #6534 |
| tests/store/tracking/test_sqlalchemy_store.py                      | @harupy      | #6534 |
| tests/tracking/test_model_registry.py                              | @harupy      | #6534 |

If you're interested in working on this issue, ping me with the file you would like to work on. Please fix only 1 ~ 3 items per pull request.

**Comment by dsgibbons** [comment]
I'm happy to look at all the mlflow/ files. Do you definitely want a separate PR for each file?

**Comment by harupy** [comment]
@dsgibbons Thanks for the help! I've updated the description. You can fix 1 ~ 3 items per pull request.

**Comment by Monkero** [comment]
Can I also participate?

**Comment by harupy** [comment]
@Monkero Absolutely!

**Comment by dsgibbons** [comment]
I'll start putting together PRs for this issue in about 12 hours @Monkero so if you want to make a start then go ahead. I'll then start fixing the ones that don't already have PRs against them, starting from the top of the list.

**Comment by harupy** [comment]
@harupy Would you mind saving the number of items you will fix for first-time contributors?

**Comment by harupy** [comment]
@Monkero Assigned the first three items to you :)

**Comment by Monkero** [comment]
Ok, I`ve done them, but do I have to write tests for it? If the answer is 'yes', then the question is 'how?'

**Comment by harupy** [comment]
@Monkero No. Did you update `mlflow/pylint_plugins/set_checker.py`?

**Comment by dsgibbons** [comment]
`mlflow/utils/search_utils.py` is contained within `pylint_plugins/set_checker.py` but is not included in the table above. `mlflow/utils/search_utils.py` still uses `set()` instead of `{}`.

**Comment by harupy** [comment]
> `mlflow/utils/search_utils.py` is contained within `pylint_plugins/set_checker.py` but is not included in the table above. `mlflow/utils/search_utils.py` still uses `set()` instead of `{}`.

I forgot to include the items I fixed in #6534. Updated the table.

**Comment by iamthen0ise** [comment]
@harupy Hi! I'd like to take some files from this task.

**Comment by harupy** [comment]
@iamthen0ise Awesome! Let me assign some items to you.

**Comment by harupy** [comment]
@iamthen0ise Done!

**Comment by harupy** [comment]
@dsgibbons Can you fix the remaining files?

**Comment by dsgibbons** [comment]
Yes @harupy

**Comment by harupy** [comment]
@dsgibbons Thanks! When you file a PR, can you remove the `IGNORED_FILES` variable?

**Comment by harupy** [comment]
@dsgibbons @iamthen0ise @Monkero Thanks for working on this issue! We have more [good-first issues.](https://github.com/mlflow/mlflow/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) Please let me know if you're interested in working on them :)

**Comment by iamthen0ise** [comment]
@harupy sure, let's try. and thank you too!

**Comment by Monkero** [comment]
Of course, thanks for helping me too, @harupy
